File: solutions/python/02/compress_coords/main.py
import bisect


def main() -> None:
    values = [10**9, 5, 10**9, 7, 5]
    uniq = sorted(set(values))
    compressed = [bisect.bisect_left(uniq, v) for v in values]
    print(uniq)
    print(compressed)


if __name__ == "__main__":
    main()

File: solutions/go/02/compress_coords/main.go
package main

import (
	"fmt"
	"sort"
)

func main() {
	values := []int{1_000_000_000, 5, 1_000_000_000, 7, 5}
	uniqSet := map[int]struct{}{}
	for _, v := range values {
		uniqSet[v] = struct{}{}
	}
	uniq := make([]int, 0, len(uniqSet))
	for v := range uniqSet {
		uniq = append(uniq, v)
	}
	sort.Ints(uniq)
	compressed := make([]int, len(values))
	for i, v := range values {
		compressed[i] = sort.SearchInts(uniq, v)
	}
	fmt.Println(uniq)
	fmt.Println(compressed)
}
